### PR TITLE
feat(router-store): Provide bindings to connect @angular/router

### LINKED ISSUE #3: Binding Angular 2 Component Router v3 w/ ngrx/store
After collaboration between angular and ngrx teams to develop new angular component v.3, it is possible to integrate this module with that?

Regards,
Emiliano

### PR DESCRIPTION
**Pull request by brandonroberts**
With the deprecation of `@ngrx/router`, this library will provide bindings to connect `@angular/router` to  `@ngrx/store`.

BREAKING CHANGES: 

Router API has changed internally

BEFORE:

Use the `routerReducer` when providing `Store`:

``` ts
  import { provideStore } from '@ngrx/store';
  import { routerReducer } from '@ngrx/router-store';


  export const storeProvider = provideStore({
    // Your reducers go here,
    router: routerReducer
  });
```

Install the bindings providers after you setup the router providers:

``` ts
  import { connectRouterToStore } from '@ngrx/router-store';

  bootstrap(App, [
    storeProvider,
    provideRouter(routes),
    connectRouterToStore()
  ]);
```

AFTER:

Use the `routerReducer` when providing the `StoreModule.provideStore` and add the `RouterStoreModule.connectRouter()` to the `@NgModule.imports`:

``` ts
  import { StoreModule } from '@ngrx/store';
  import { routerReducer, RouterStoreModule } from '@ngrx/router-store';

  @NgModule({
    imports: [
      BrowserModule,
      StoreModule.provideStore({ router: routerReducer }),
      RouterStoreModule.connectRouter()
    ],
    bootstrap: [ AppComponent ]
  })
  export class AppModule {
  }
```

BEFORE:

``` ts
import {routerActions} from '@ngrx/store';

store.dispatch(routerActions.go('/path', 'query=string'));
store.dispatch(routerActions.replace('/path', 'query=string'));
store.dispatch(routerActions.search('query=string'));
```

AFTER:

``` ts
import {routerActions} from '@ngrx/store';

store.dispatch(routerActions.go('/path', { query: 'string' ));
store.dispatch(routerActions.replace('/path', { query: 'string' ));
store.dispatch(routerActions.search({ query: 'string' ));
```

Closes #3

### CODE REVIEW

File: lib/connect.ts [hunk 1]
```diff
@@ -1,49 +1,74 @@
...
 
       switch (action.type) {
         case routerActions.GO:
-          router.go(path, query);
+          router.navigate(commands, { queryParams });
           break;
 
         case routerActions.REPLACE:
-          router.replace(path, query);
+          router.navigate(commands, { queryParams });
```
Comment (MikeRyanDev) [inline lib/connect.ts]: What does the API look like for navigating silently in the next release?

Comment (brandonroberts) [reply lib/connect.ts]: Silent navigation looks like
`router.navigate(['/path'], { skipLocationChange: true })`

Replacing the state looks like
`router.navigate(['/path'], { replaceUrl: true })`

Comment (MikeRyanDev) [reply lib/connect.ts]: Understood. Can we go ahead and pass those options in here knowing that in RC5 they won't actually do anything?

Comment (MikeRyanDev) [reply lib/connect.ts]: We will also want to add an action for navigating without location change. That wasn't really supported in @ngrx/router.

Comment (brandonroberts) [reply lib/connect.ts]: Ok. `routerActions.SHOW` for navigating without a location change?

Comment (MikeRyanDev) [reply lib/connect.ts]: That sounds reasonable to me.

**Comment by brandonroberts** [comment]
@MikeRyan52 ready for another review. Docs and tests have been updated.

**Comment by MikeRyanDev** [comment]
LGTM. Will try to publish a beta tonight. Need to replace the current build infra with ngc.

**Comment by mhodgson** [comment]
Just a quick comment, but it would be very useful to store the entire `router.snapshot` (ActivatedRouteSnapshot) in the state as well as just the simple URL. This makes getting parameters, data, and resolves out of the router state much easier.

**Comment by brandonroberts** [comment]
@mhodgson The `router.snapshot` has references to components, modules and many other things, so it could not serialized into json in a clean way.

**Comment by BrainCrumbz** [comment]
Could you please drop a line here when a new package with this change is released? Or any other visible notification as well. Thanks a lot!

**Comment by MikeRyanDev** [comment]
@BrainCrumbz It has been released as v1.0.0

**Pull request merged**